src/mesh/X3D_Draco_Mesh_Reader.hh

```diff
@@ -114,6 +118,12 @@ public:
     return x3d_coord_map.at(static_cast<int>(node + 1));
   }
 
+  // matid data
+  std::string get_matid(size_t cell) const {
+    Check(cell + 1 < INT_MAX);
+    return x3d_matids.at(cell);
```

Comment:
so x3d_matids stores a separate string for each cell? That sounds expensive, though I guess mesh setup isn't that big of a player overall

Reply:
It does; but since the entire file is broken into individual strings for processing, it's basically noise.